internal/cmd/postgresflex/instance/clone/clone.go

```diff
@@ -0,0 +1,154 @@
+package clone
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stackitcloud/stackit-cli/internal/pkg/args"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/confirm"
+	cliErr "github.com/stackitcloud/stackit-cli/internal/pkg/errors"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/examples"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/flags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/postgresflex/client"
+	postgresflexUtils "github.com/stackitcloud/stackit-cli/internal/pkg/services/postgresflex/utils"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/spinner"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/utils"
+
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-sdk-go/services/postgresflex"
+	"github.com/stackitcloud/stackit-sdk-go/services/postgresflex/wait"
+)
+
+const (
+	instanceIdArg = "INSTANCE_ID"
+
+	storageClassFlag      = "storage-class"
+	storageSizeFlag       = "storage-size"
+	recoveryTimestampFlag = "recovery-timestamp"
+	recoveryDateFormat    = "2023-04-17T09:28:00+00:00"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+
+	InstanceId   string
+	StorageClass *string
+	StorageSize  *int64
+	RecoveryDate *string
+}
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("clone %s", instanceIdArg),
+		Short: "Clones a PostgreSQL Flex instance",
+		Long:  "Clones a PostgreSQL Flex instance from a selected point in time.",
```

Comment:
I think here would also be a good place to eleborate a bit more, saying the clone will be an independent instance with the same settings as the original instance unless the flags are specified etc

Reply:
What do you think?